tests/unit_tests/agent/node/test_gen_semantic_model_agentic_node.py

```
@@ -0,0 +1,266 @@
+    build_tool_then_response,
+)
+
+# ---------------------------------------------------------------------------
+# Initialization Tests
+# ---------------------------------------------------------------------------
+
+
+class TestGenSemanticModelAgenticNodeInit:
+    """Tests for GenSemanticModelAgenticNode initialization."""
+
+    def test_semantic_model_init(self, real_agent_config, mock_llm_create):
```

Ruff (0.15.0)

[warning] Unused method argument: `mock_llm_create`

(ARG002)